src/csharp/Microsoft.Spark/Sql/WorkerFunction.cs

```diff
@@ -81,6 +81,62 @@ internal WorkerFuncChainHelper(ExecuteDelegate inner, ExecuteDelegate outer)
         }
     }
 
+    /// <summary>
+    /// Function for Grouped Map Vector UDFs using the Apache Arrow format.
+    /// </summary>
+    internal sealed class ArrowGroupedMapWorkerFunction : WorkerFunction
+    {
+        /// <summary>
+        /// A delegate to invoke a Grouped Map Vector UDF.
+        /// </summary>
+        /// <param name="input">The input data frame.</param>
+        /// <returns>The resultant data frame.</returns>
+        internal delegate RecordBatch ExecuteDelegate(RecordBatch input);
+
+        internal ArrowGroupedMapWorkerFunction(ExecuteDelegate func)
+        {
+            Func = func;
+        }
+
+        internal ExecuteDelegate Func { get; }
+
+        /// <summary>
+        /// Used to chain functions.
+        /// </summary>
+        internal static ArrowGroupedMapWorkerFunction Chain(
+            ArrowGroupedMapWorkerFunction innerWorkerFunction,
+            ArrowGroupedMapWorkerFunction outerWorkerFunction)
+        {
+            return new ArrowGroupedMapWorkerFunction(
+                new WorkerFuncChainHelper(
+                    innerWorkerFunction.Func,
+                    outerWorkerFunction.Func).Execute);
+        }
+
+        private class WorkerFuncChainHelper
```

Comment:
I don't know how to chain Grouped Map UDFs. Do you know how?

I was thinking something like:

```C#
            rows = df.GroupBy("age")
                .Apply(
                    new StructType(new[]
                    {
                        new StructField("age", new IntegerType()),
                        new StructField("nameCharCount", new IntegerType())
                    }),
                    batch => CountCharacters(batch))
                .GroupBy("nameCharCount")
                    .Apply(new StructType(new[]
                        {
                            new StructField("nameCharCount", new IntegerType()),
                            new StructField("ageSum", new IntegerType())
                        }),
                        batch => SumAges(batch))
                .Collect()
                .ToArray();
```

But that doesn't do it. I assume because you need to execute the first `Apply` in order to get the groups for the second.

If you can think of how to do this, I will add it.

For now, I could remove the "Multiple" functionality, and throw if we get more than 1 Grouped Map UDF. That way we would know if it ever happens in the wild.

Reply:
> For now, I could remove the "Multiple" functionality, and throw if we get more than 1 Grouped Map UDF.

Yes, this is the right behavior: https://github.com/apache/spark/blob/ec6d0c9ab2a3cab586e73def9167a120780a2fbe/python/pyspark/worker.py#L225

Thanks!